arch/common/semihost.c

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2022, Commonwealth Scientific and Industrial Research
+ * Organisation (CSIRO) ABN 41 687 119 230.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <zephyr/kernel.h>
+#include <zephyr/arch/common/semihost.h>
+
+struct _semihost_poll_in_args {
```

Comment:
Please do not add any new identifiers that start with `_` (i.e. reserved identifiers).